test/games/strategy/engine/data/UnitCollectionTest.java

```diff
@@ -0,0 +1,386 @@
+    uc.addAllUnits(getUnitsOfDefaultPlayerOfUnitTypeTwo());
+    assertThat(uc.size(), is(equalTo(unitCountDefaultPlayerUnitTypeOne + unitCountDefaultPlayerUnitTypeTwo)));
+  }
+
+  @Test
+  public void isEmpty() {
+    assertThat(uc.isEmpty(), is(equalTo(true)));
+    uc.addAllUnits(getUnitsOfDefaultPlayerOfUnitTypeOne());
+    assertThat(uc.isEmpty(), is(equalTo(false)));
+  }
+
+  @Test
+  public void getUnits() {
+    final Collection<Unit> unitsOfOtherPlayerOfUnitTypeOne = getUnitsOfOtherPlayerOfUnitTypeOne();
+    uc.addAllUnits(unitsOfOtherPlayerOfUnitTypeOne);
+    assertThat(uc.getUnits().containsAll(unitsOfOtherPlayerOfUnitTypeOne), is(equalTo(true)));
+  }
+
+  @Test
+  public void getPlayersWithUnits() {
+    assertThat(uc.getPlayersWithUnits().size(), is(equalTo(0)));
+    addAllDefaultPlayerUnits();
+    assertThat(uc.getPlayersWithUnits().size(), is(equalTo(1)));
+    addAllOtherPlayerUnits();
+    assertThat(uc.getPlayersWithUnits().size(), is(equalTo(2)));
+  }
+
+  @Test
+  public void getPlayerUnitCounts() {
+    addAllPlayerUnits();
+    final IntegerMap<PlayerID> playerUnitCounts = uc.getPlayerUnitCounts();
+    assertThat(playerUnitCounts.getInt(defaultPlayerID), is(equalTo(unitCountDefaultPlayer)));
+    assertThat(playerUnitCounts.getInt(otherPlayerID), is(equalTo(unitCountOtherPlayer)));
+  }
+
+  @Test
+  public void hasUnitsFromMultiplePlayers() {
+    addAllDefaultPlayerUnits();
+    assertThat(uc.hasUnitsFromMultiplePlayers(), is(equalTo(false)));
+    addAllOtherPlayerUnits();
+    assertThat(uc.hasUnitsFromMultiplePlayers(), is(equalTo(true)));
+  }
+
+  @Test
+  public void getHolder() {
+    assertThat(uc.getHolder(), is(equalTo(defaultPlayerID)));
+  }
+
+  @Test
+  public void allMatch() {
+    uc.addAllUnits(getUnitsOfDefaultPlayerOfUnitTypeOne());
+    assertThat(uc.allMatch(Matches.unitIsOfType(unitTypeOne)), is(equalTo(true)));
+    uc.addAllUnits(getUnitsOfDefaultPlayerOfUnitTypeTwo());
+    assertThat(uc.allMatch(Matches.unitIsOfType(unitTypeOne)), is(equalTo(false)));
+  }
+
+  @Test
+  public void someMatch() {
+    uc.addAllUnits(getUnitsOfDefaultPlayerOfUnitTypeOne());
+    assertThat(uc.someMatch(Matches.unitIsOfType(unitTypeOne)), is(equalTo(true)));
+    assertThat(uc.someMatch(Matches.unitIsOfType(unitTypeTwo)), is(equalTo(false)));
+    uc.addAllUnits(getUnitsOfDefaultPlayerOfUnitTypeTwo());
+    assertThat(uc.someMatch(Matches.unitIsOfType(unitTypeOne)), is(equalTo(true)));
+    assertThat(uc.someMatch(Matches.unitIsOfType(unitTypeTwo)), is(equalTo(true)));
+  }
+
+  @Test
+  public void countMatches() {
+    addAllDefaultPlayerUnits();
+    assertThat(uc.countMatches(Matches.unitIsOfType(unitTypeOne)), is(equalTo(unitCountDefaultPlayerUnitTypeOne)));
+    assertThat(uc.countMatches(Matches.unitIsOfType(unitTypeTwo)), is(equalTo(unitCountDefaultPlayerUnitTypeTwo)));
+  }
+
+  @Test
+  public void getMatches() {
+    addAllDefaultPlayerUnits();

```

Comment:
Before modifying the test data, woudl be good to validate the state of the test data. Same note about avoiding side effects as well, avoid the `void` return value on these helper methods (`addAllDefaultPlayerUnits()` -> `getAllDefaultPlayerUnits`)